src/emu/validity.cpp

```diff
@@ -1822,6 +1823,14 @@ void validity_checker::validate_inputs()
 				for (ioport_setting &setting : field.settings())
 					if (!setting.condition().none())
 						validate_condition(setting.condition(), device, port_map);
+
+				// verify natural keyboard codes
+				for (int which = 0; which < natural_keyboard::SHIFT_STATES; which++)
```

Comment:
This shouldn't be using `natkeyboard.h` - that's an implementation detail.  The message could be better, something like "Field '%s' has non-character U+%04X in `PORT_CHAR` %d\n".